Permanent/ViewControllers/TagManagementViewController.swift

```diff
@@ -0,0 +1,186 @@
+        
+        collectionView.backgroundColor = .clear
+        collectionView.indicatorStyle = .white
+        
+        let flowLayout = UICollectionViewFlowLayout()
+        collectionView.collectionViewLayout = flowLayout
+    }
+    
+    func updateTagsNumber() {
+        guard let tagsNumber = viewModel?.sortedTags.count else {
+            archiveTitleTagsNbrLabel.text = nil
+            return
+        }
+        
+        archiveTitleTagsNbrLabel.text = "(\(String(tagsNumber)))"
+    }
+}
+
+// MARK: - FABView Delegate
+extension TagManagementViewController: FABViewDelegate {
+    func didTap() {
+        guard let actionSheet = UIViewController.create(
+            withIdentifier: .fabActionSheet,
+            from: .main
+        ) as? FABActionSheet else {
+            showAlert(title: .error, message: .errorMessage)
+            return
+        }
+
+        //actionSheet.delegate = self
+        navigationController?.display(viewController: actionSheet, modally: true)
+    }
+}
+
+// MARK: - SearchBar Delegate
+extension TagManagementViewController: UISearchBarDelegate {
+    func searchBar(_ searchBar: UISearchBar, textDidChange searchText: String) {
+
+        let currentTagVO: [TagVO] = viewModel?.tags.map({ (item) -> TagVO in
+            return item
+        }) ?? []
+
+        viewModel?.sortedTags = searchText.isEmpty ? currentTagVO  : currentTagVO.filter({ (tag) -> Bool in
+            return tag.tagVO.name?.lowercased().contains(searchText.lowercased()) ?? false
+        })
+    }
+
+    func searchBarSearchButtonClicked(_ searchBar: UISearchBar) {
+        self.view.endEditing(true)
+    }
+    
+    func searchBar(_ searchBar: UISearchBar, shouldChangeTextIn range: NSRange, replacementText text: String) -> Bool {
+        if range.location >= 16 { return false }
+        return true
+    }
+}
+
+// MARK: - CollectionView Delegate
+extension TagManagementViewController: UICollectionViewDelegate {
+    
+}
+
+// MARK: - CollectionView DataSource
+extension TagManagementViewController: UICollectionViewDataSource {
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        return viewModel?.sortedTags.count ?? 0
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        let cell = collectionView.dequeueReusableCell(withReuseIdentifier: ArchiveSettingsTagCollectionViewCell.identifier, for: indexPath) as! ArchiveSettingsTagCollectionViewCell
+        
+        if let tagName = viewModel?.sortedTags[indexPath.item] {
+            cell.configure(tagName: tagName.tagVO.name)
+            
+            cell.rightSideButtonAction = {
+                self.showSpinner()
```

Comment:
Let's create a "isLoading" variable on the viewModel. The viewModel should post a notification whenever the "isLoading" state changes. 
This ViewController should listen to that notification and show/hide the spinner based on the "isLoading" state